qiskit/tools/wigner_function_visualisation.py

```diff
@@ -0,0 +1,264 @@
+"""
+
+Wigner Function Visualization Module
+
+Description:
+    This module contains functions for calculating and plotting the Wigner
+    function. The ways the Wigner function can be calculated includes:
+    - The equal angle slice bloch sphere plot of an arbitrary state
+      described by a density matrix or state vector.
+    - An array of points in phase space to be plotted as a curve
+    - A single point in phase space
+    - An NxN array plotted as a plaquette
+
+References:
+    [1] T. Tilma, M. J. Everitt, J. H. Samson, W. J. Munro,
+and K. Nemoto,
+        Phys. Rev. Lett. 117, 180401 (2016).
+    [2] R. P. Rundle, P. W. Mills, T. Tilma, J. H. Samson, and
+M. J. Everitt,
+        Phys. Rev. A 96, 022117 (2017).
+"""
+
+import math
+import numpy as np
+import matplotlib
+from matplotlib import pyplot as plt
+from matplotlib import cm
+from mpl_toolkits.mplot3d import Axes3D
+
+
+def wigner_function(rho, res=100):
+
+    """
+    Plot the equal angle slice spin Wigner function of an arbitrary
+    quantum state
+
+    Args:
+        rho (np.matrix[[complex]]): - Matrix of 2**n x 2**n complex
+                                        numbers
+                                    - State Vector of 2**n x 1 complex
+                                        numbers
+        res (int) : number of theta and phi values in meshgrid
+                    on sphere (creates a res x res grid of points)
+    Returns:
+        none: plot is shown with matplotlib on the screen and expression for
+            Wigner function printed
+        
+    """
+
```

Comment:
This function breaks if the input is a list. This can be fixed by adding
```python
def wigner_function(state, res=100):
    ## ....
    state = np.array(state)
    if state.ndim == 1:
        state = np.outer(state, state)
```